### ISSUE TITLE
validation order is messed up

### ISSUE THREAD

**Issue by TrejGun** [Automattic/mongoose#4861]
**Do you want to request a *feature* or report a *bug*?**
bug

**If the current behavior is a bug, please provide the steps to reproduce.**

```js
var q = require("q");
var mongoose = require("mongoose");
mongoose.set("debug", true);
mongoose.Promise = q.Promise;

var db1 = mongoose.createConnection();
db1.open("mongodb://localhost/issue-1234-1");

var X = db1.model("X", new mongoose.Schema({
	a: {
		type: Number,
		min: [1, "a"]
	},
	b: {
		type: Number,
		min: [1, "b"]
	}
}));


X.create({
	a: 0,
	b: 0
})
	.catch(function (e1) {
		console.log(Object.keys(e1.errors));
		return X.create({
			a: 10,
			b: 10
		})
			.then(function (x) {
				Object.assign(x, {a: 0, b: 0});
				return x.save()
					.catch(function (e2) {
						console.log(Object.keys(e2.errors));
					});
			});
	})
	.then(function () {
		return X.remove().then(function () {
			process.exit(0);
		});
	});
```
**What is the current behavior?**
validation runs in different directions 
```
[ 'b', 'a' ] // create runs from last to first
[ 'a', 'b' ] // save
```

**What is the expected behavior?**
validation should give same output for both functions, preferably as save works
```
[ 'a', 'b' ] // create
[ 'a', 'b' ] // save
```

**Please mention your node.js, mongoose and MongoDB version.**
mongoose 4.7.5
➜ node -v
v7.0.0
➜ npm -v
4.0.2
➜ mongo --version
MongoDB shell version v3.4.0

**Comment by vkarpov15** [comment]
#2749 strikes again. Just use the `retainKeyOrder` option (which is deprecated, the expected behavior will be the default in 5.0)

```javascript
var X = db1.model("X", new mongoose.Schema({
        a: {
                type: Number,
                min: [1, "a"]
        },
        b: {
                type: Number,
                min: [1, "b"]
        }
}, { retainKeyOrder: true }));
```

**Comment by goffjira** [comment]
Guys, Is it got fixed? I still see this problem for mongoose@5.12.12  where the validation errors being put to 'errors' property are listed there in reverse order when compare which how you define them in schema. to be more specific see code below

//this is how you define them in schema, username comes first then password.
`const schema = new mongoose.Schema(
  {
    username: {
        type: String,
        required: [true, "username is required"]},
    password: {
        type: String,
        required: [true, "password is required"]},
}

//this is how you get it, when the validation being throw
errors: {
        password: ValidatorError: password is required,
        username: ValidatorError: username is required,
}

I tried with 'retainKeyOrder' option which does'not work and obviously deprecated any thought?




`
